### PR TITLE
Add .NET Framework compat package

### PR DESCRIPTION
**Pull request by weshaggard**
This meta-package will make it easier for .NET Core developers to add a set of
library packages to help them port code from .NET Framework to .NET Core.

resolves https://github.com/dotnet/corefx/issues/21664. 

PTAL @ericstj @Petermarcu @terrajobst

### CODE REVIEW

File: pkg/descriptions.json [hunk 1]
```diff
@@ -51,6 +51,11 @@
     "CommonTypes": []
   },
   {
+    "Name": "Microsoft.NETFramework.Compatibility",
+    "Description": "References a number of .NET Core libraries that are intended to provide support for more APIs that exists on .NET Framework.",
```
Comment (danmosemsft) [inline pkg/descriptions.json]: "that exist"

File: pkg/Microsoft.NETFramework.Compatibility/Microsoft.NETFramework.Compatibility.pkgproj [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    <LibraryPackage Include="System.ComponentModel.Annotations" />
+    <LibraryPackage Include="System.Composition" />
+    <LibraryPackage Include="System.Composition.AttributedModel" />
+    <LibraryPackage Include="System.Composition.Convention" />
+    <LibraryPackage Include="System.Composition.Hosting" />
+    <LibraryPackage Include="System.Composition.Runtime" />
+    <LibraryPackage Include="System.Composition.TypedParts" />
+    <LibraryPackage Include="System.Configuration.ConfigurationManager" />
+    <LibraryPackage Include="System.Data.SqlClient" />
+    <LibraryPackage Include="System.DirectoryServices" />
```
Comment (danmosemsft) [inline pkg/Microsoft.NETFramework.Compatibility/Microsoft.NETFramework.Compatibility.pkgproj]: I'm not sure what quality level the directory services packages are in. They're not currently shipped right?

Comment (weshaggard) [inline pkg/Microsoft.NETFramework.Compatibility/Microsoft.NETFramework.Compatibility.pkgproj]: They are currently in pre-release but they are on the list to ship. We can exclude them from here for now if you have concerns with them.

File: pkg/Microsoft.NETFramework.Compatibility/Configurations.props [hunk 3]
```diff
@@ -0,0 +1,8 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <BuildConfigurations>
+      netcoreapp;
```
Comment (ericstj) [inline pkg/Microsoft.NETFramework.Compatibility/Configurations.props]: Really?  Typically I mark the packages for "package" configuration which only builds during AllConfiguration leg.  I guess since it's only a meta-package it doesn't matter so much, but a lot of the things its referencing only build in the AllConfiguration leg.

File: pkg/Microsoft.Private.PackageBaseline/packageIndex.json [hunk 4]
```diff
@@ -54,7 +54,8 @@
       "StableVersions": [
         "4.0.1",
         "4.0.0",
-        "4.3.0"
+        "4.3.0",
+        "4.4.0"
```
Comment (ericstj) [inline pkg/Microsoft.Private.PackageBaseline/packageIndex.json]: If you're going to do this, can you make sure you do it holistically with all the stable changes that were made in the release branch? That should be done in a separate commit.  At the same time you need to make sure you update the harvest project so that we download these for harvesting.

File: pkg/Microsoft.NETFramework.Compatibility/Microsoft.NETFramework.Compatibility.pkgproj (hunk 2 above)
Comment (danmosemsft) [inline pkg/Microsoft.NETFramework.Compatibility/Microsoft.NETFramework.Compatibility.pkgproj]: @tquerec what is the quality level of each of these? 
System.DirectoryServices
System.DirectoryServices.AccountManagement
System.DirectoryServices.Protocols

File: pkg/Microsoft.NETFramework.Compatibility/Configurations.props (hunk 3 above)
Comment (weshaggard) [inline pkg/Microsoft.NETFramework.Compatibility/Configurations.props]: Yeah I could make this just a package configuration as well.

**Comment by weshaggard** [comment]
>Please split up the index update and make sure its a complete update from release.

For this PR I decided to not rely on Baselining and instead just specify the versions. I will update the baselines as a separate change.

**Comment by weshaggard** [comment]
@Petermarcu @terrajobst can you guys OK the package name?

**Comment by weshaggard** [comment]
@terrajobst @Petermarcu I'm going to move forward with my current name. If you would like another name please file an issue and we can rename it.

File: pkg/Microsoft.NETFramework.Compatibility/Microsoft.NETFramework.Compatibility.pkgproj [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    <LibraryPackage Include="System.Security.Cryptography.Cng" />
+    <LibraryPackage Include="System.Security.Cryptography.Pkcs" />
+    <LibraryPackage Include="System.Security.Cryptography.ProtectedData" />
+    <LibraryPackage Include="System.Security.Cryptography.Xml" />
+    <LibraryPackage Include="System.Security.Permissions" />
+    <LibraryPackage Include="System.Security.Principal.Windows" />
+    <LibraryPackage Include="System.ServiceProcess.ServiceController" />
+    <LibraryPackage Include="System.Text.Encoding.CodePages" />
+    <LibraryPackage Include="System.Threading.AccessControl" />
+    <LibraryPackage Include="System.Threading.Tasks.Dataflow">
```
Comment (ericstj) [inline pkg/Microsoft.NETFramework.Compatibility/Microsoft.NETFramework.Compatibility.pkgproj]: Surprised to see dataflow here.  It was actually never part of desktop.

File: pkg/Microsoft.NETFramework.Compatibility/Microsoft.NETFramework.Compatibility.pkgproj [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  </PropertyGroup>
+
+  <ItemDefinitionGroup>
+    <LibraryPackage>
+      <Version>4.5.0</Version>
+    </LibraryPackage>
+  </ItemDefinitionGroup>
+
+  <ItemGroup>
+    <LibraryPackage Include="System.CodeDom" />
```
Comment (ericstj) [inline pkg/Microsoft.NETFramework.Compatibility/Microsoft.NETFramework.Compatibility.pkgproj]: I wonder if you could do something in the build to try and catch an incomplete list?  For example, any package that has a full-facade for desktop, or anything that is the target of one of the desktop compat facades (system, system.core, etc)

**Comment by mellinoe** [comment]
Should we add System.Drawing.Common to this list?

File: pkg/Microsoft.NETFramework.Compatibility/Microsoft.NETFramework.Compatibility.pkgproj (hunk 5 above)
Comment (weshaggard) [inline pkg/Microsoft.NETFramework.Compatibility/Microsoft.NETFramework.Compatibility.pkgproj]: True DataFlow is OOB and should be removed.

File: pkg/Microsoft.NETFramework.Compatibility/Microsoft.NETFramework.Compatibility.pkgproj (hunk 6 above)
Comment (weshaggard) [inline pkg/Microsoft.NETFramework.Compatibility/Microsoft.NETFramework.Compatibility.pkgproj]: I was trying to come up with a way to validate not only that it is complete but also things had matching versions but I couldn't come up with anything clever. Let me know if you have any ideas.

**Comment by weshaggard** [comment]
>Should we add System.Drawing.Common to this list?

Yes it probably should be added now. I don't know why that one didn't show up in my original list, perhaps my list was before your addition of the package for that.
